src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -26,6 +31,35 @@ public static IServiceCollection AddSignalRServiceManager(this IServiceCollectio
             services.AddSingleton<ServiceOptionsSetup>()
                     .AddSingleton<IConfigureOptions<ServiceOptions>>(sp => sp.GetService<ServiceOptionsSetup>())
                     .AddSingleton<IOptionsChangeTokenSource<ServiceOptions>>(sp => sp.GetService<ServiceOptionsSetup>());
+
+            //add dependencies for persistent mode only
+            services.AddSingleton<IServerNameProvider, DefaultServerNameProvider>()
+                .AddSingleton<ConnectionFactory>()
+                .AddSingleton<IConnectionFactory>(sp =>
+                {
+                    var productInfo = sp.GetRequiredService<IOptions<ServiceManagerContext>>().Value.ProductInfo;
+                    var defaultConnectionFactory = sp.GetRequiredService<ConnectionFactory>();
+                    return new ManagementConnectionFactory(productInfo, defaultConnectionFactory);
+                })
+                .AddSingleton<IServiceProtocol, ServiceProtocol>()
+                .AddSingleton<IClientConnectionManager, ClientConnectionManager>()
+                .AddSingleton<IClientConnectionFactory, ClientConnectionFactory>()
+                .AddSingleton<IServiceConnectionFactory>(sp =>
+                    ActivatorUtilities.CreateInstance<ServiceConnectionFactory>(sp, (ConnectionDelegate)((connectionContext) => Task.CompletedTask)))
+                .AddSingleton<MultiEndpointConnectionContainerFactory>()
+                .AddSingleton<IServiceEndpointManager, ServiceEndpointManager>()
+                .AddSingleton<IConfigureOptions<HubOptions>, ManagementHubOptionsSetup>()
+                .AddSingleton(typeof(IServiceConnectionManager<>), typeof(ServiceConnectionManager<>));
+
+            services.AddSignalRCore();
```

Comment:
Is `AddAzureSignalR` first then replace those with Management specific ones clearer?

Reply:
Good idea.